src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -146,9 +147,296 @@ int dump_parsed_trace_events(PyObject* trace_events, FILE* fptr)
+    PyObject* diffed_args = PyDict_New();
+    PyObject* ts_keys = PyDict_Keys(counter_args);
+    PyObject* ts = NULL;
+    PyObject* cur_diffed_arg = NULL;
+    PyObject* cur_counter_arg = NULL;
+    PyObject* arg_key_list = NULL;
+    PyObject* counter_arg_key = NULL;
+    PyObject* counter_arg_value = NULL;
+    PyObject* cached_arg_value = NULL;
+    PyObject* overflowed_num_string = NULL;
+    Py_ssize_t ppos = 0;
+    uint64_t ts_key_count = 0;
+    uint64_t arg_nums = 0;
+    // sort the args by timestamp so we can diff
+    if (!ts_keys || !PyList_Check(ts_keys)) {
+        PyErr_SetString(PyExc_ValueError, "failed to get timestamp list");
+        goto clean_exit;
+    }
+    ts_key_count = PyList_GET_SIZE(ts_keys);
+    if (PyList_Sort(ts_keys) == -1) {
+        goto clean_exit;
+    }
+    // do diffing between two timestamps and store the result
+    for (uint64_t i = 0; i < ts_key_count; i++) {
+        ts = PyList_GET_ITEM(ts_keys, i);
+        cur_counter_arg = PyDict_GetItem(counter_args, ts);
+        cur_diffed_arg = PyDict_New();
+        ppos = 0;
+        while (PyDict_Next(cur_counter_arg, &ppos, &counter_arg_key, &counter_arg_value)) {
+            cached_arg_value = PyDict_GetItem(cached_args_dict, counter_arg_key);
+            if (!cached_arg_value) {
+                PyDict_SetItem(cached_args_dict, counter_arg_key, counter_arg_value);
+                PyDict_SetItem(cur_diffed_arg, counter_arg_key, counter_arg_value);
+            } else{
+                int compare_result = PyObject_RichCompareBool(cached_arg_value, counter_arg_value, Py_EQ);
+                if (compare_result == -1) {
+                    // compare error 
+                    goto clean_exit;
+                } else if (compare_result == 0) {
+                    // if value is not equal in last timestamp, store the newest value
+                    PyDict_SetItem(cached_args_dict, counter_arg_key, counter_arg_value);
+                    PyDict_SetItem(cur_diffed_arg, counter_arg_key, counter_arg_value);
+                }
+            }
+        }
+        ppos = 0;
+        while (PyDict_Next(cached_args_dict, &ppos, &counter_arg_key, &cached_arg_value)) {
+            counter_arg_value = PyDict_GetItem(cur_counter_arg, counter_arg_key);
+            if (!counter_arg_value && cached_arg_value != Py_None) {
+                PyDict_SetItem(cached_args_dict, counter_arg_key, Py_None);
+                PyDict_SetItem(cur_diffed_arg, counter_arg_key, Py_None);
+            }
+        }
+        PyDict_SetItem(diffed_args, ts, cur_diffed_arg);
+        Py_DECREF(cur_diffed_arg);
+    }
+    // write all the arg keys
+    arg_nums = PyDict_Size(cached_args_dict);
+    fwrite(&arg_nums, sizeof(uint64_t), 1, fptr);
+    arg_key_list = PyDict_Keys(cached_args_dict);
+    if (!arg_key_list) {
+        PyErr_SetString(PyExc_ValueError, "failed to get arg name list");
+        goto clean_exit;
+    }
+    for (uint64_t i = 0; i < arg_nums; i++) {
```

Comment:
Same for the comment here. It's simpler, but it would be nice to have some comments to explain what you are doing. Maybe just a single sentence saying you are wrting all the name for all arguments appeared.